builtin/logical/ssh/path_config_issuers.go

```diff
@@ -0,0 +1,150 @@
+			logical.ReadOperation: &framework.PathOperation{
+				Callback: b.pathReadDefaultIssuerHandler,
+				DisplayAttrs: &framework.DisplayAttributes{
+					OperationSuffix: "read",
+				},
+				Responses: configIssuerSchema,
+			},
+			logical.UpdateOperation: &framework.PathOperation{
+				Callback: b.pathWriteDefaultIssuerHandler,
+				DisplayAttrs: &framework.DisplayAttributes{
+					OperationVerb: "write",
+				},
+				Responses: configIssuerSchema,
+			},
+		},
+
+		HelpSynopsis:    pathConfigIssuersSyn,
+		HelpDescription: pathConfigIssuersDesc,
+	}
+}
+
+func (b *backend) pathReadDefaultIssuerHandler(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	sc := b.makeStorageContext(ctx, req.Storage)
+	config, err := sc.getIssuersConfig()
+	if err != nil {
+		return handleStorageContextErr(err, "error loading issuers configuration")
+	}
+
+	if len(config.DefaultIssuerID) == 0 {
+		return logical.ErrorResponse("no default issuer currently configured"), nil
+	}
+
+	return &logical.Response{
+		Data: map[string]interface{}{
+			defaultRef: config.DefaultIssuerID,
+		},
+	}, nil
+}
+
+func (b *backend) pathWriteDefaultIssuerHandler(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	// Since we're planning on updating issuers here, grab the lock so we've
+	// got a consistent view.
+	b.issuersLock.Lock()
+	defer b.issuersLock.Unlock()
+
+	// Use the transaction storage if there's one.
+	if txnStorage, ok := req.Storage.(logical.TransactionalStorage); ok {
+		txn, err := txnStorage.BeginTx(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		defer txn.Rollback(ctx)
+		req.Storage = txn
+	}
+
+	sc := b.makeStorageContext(ctx, req.Storage)
+
+	// Validate the new default reference.
+	newDefault := getDefaultRef(d)
+	if len(newDefault) == 0 || newDefault == defaultRef {
+		return logical.ErrorResponse("invalid issuer specification; must be non-empty and can't be 'default'."), nil
+	}
+	parsedIssuer, err := sc.resolveIssuerReference(newDefault)
+	if err != nil {
+		return handleStorageContextErr(err, "error resolving issuer reference")
+	}
+
+	// Update the config
+	config, err := sc.getIssuersConfig()
+	if err != nil {
+		return handleStorageContextErr(err, "unable to fetch existing issuers configuration")
+	}
+
+	oldDefault := config.DefaultIssuerID
+	config.DefaultIssuerID = parsedIssuer
+
+	if err := sc.setIssuersConfig(config); err != nil {
+		return handleStorageContextErr(err, "error updating issuer configuration")
+	}
+
+	// Commit our transaction if we created one!
+	if txn, ok := req.Storage.(logical.Transaction); ok {
+		if err := txn.Commit(ctx); err != nil {
+			return nil, err
+		}
+	}
+
+	response := &logical.Response{
+		Data: map[string]interface{}{
+			defaultRef: config.DefaultIssuerID,
+		},
+	}
+	if len(oldDefault) != 0 {
+		warningMessage := fmt.Sprintf("Previous default (%s) has been updated with the issuer '%s'", oldDefault, newDefault)
```

Comment:
I'm curious why this warning was added?

Reply:
Hmm, ok, we'll keep it for now. I guess it is more useful in SSH than PKI since the default behavior is a touch more complex over here.